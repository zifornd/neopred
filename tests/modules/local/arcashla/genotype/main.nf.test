nextflow_process {

    name "Test Process ARCASHLA_GENOTYPE"
    script "modules/local/arcashla/genotype/main.nf"
    process "ARCASHLA_GENOTYPE"

    tag "modules"
    tag "modules_local"
    tag "arcas_hla/genome_generate"

    test("ARCASHLA_GENOTYPE") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results" //params.outputDir
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = Channel.of([
                    [id: "test_1"],
                    [ file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkIfExists: true) ],
                    [file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll (
            //{assert path("$outdir/results").exists()},
            {assert process.success},
	        {assert process.out.alignment_p},
            { assert process.out.alignment_p},
	        {assert process.out.gt_log},
	        {assert process.out.gt_json},
	        {assert process.out.genes_json},
	        {assert process.out.versions},
            {assert snapshot(process.out)}
            )
        }
    }

}

