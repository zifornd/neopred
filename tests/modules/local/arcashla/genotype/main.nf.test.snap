{
    "ARCASHLA_GENOTYPE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.alignment.p:md5,f3182ccbf252b9e0065964459aaa2a73"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.genotype.log:md5,0a32ab90b559a5b8046959b7b21f7ef7"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.genotype.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.genes.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                    ]
                ],
                "4": [
                    "versions.yml:md5,dd5861b7e77f6ebf72239f4f74a0ce6e"
                ],
                "alignment_p": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.alignment.p:md5,f3182ccbf252b9e0065964459aaa2a73"
                    ]
                ],
                "genes_json": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.genes.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                    ]
                ],
                "gt_json": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.genotype.json:md5,99914b932bd37a50b983c5e7c90ae93b"
                    ]
                ],
                "gt_log": [
                    [
                        {
                            "id": "test_1"
                        },
                        "test_1.genotype.log:md5,0a32ab90b559a5b8046959b7b21f7ef7"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,dd5861b7e77f6ebf72239f4f74a0ce6e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-07T07:01:59.711008"
    }
}